lms/djangoapps/certificates/migrations/0009_certificategenerationcoursesetting_language_self_generation.py

```diff
@@ -0,0 +1,40 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from django.apps import apps
+from django.db import migrations, models
+from django.db.models import F
+
+def copy_field(apps, schema_editor):
+    CertificateGenerationCourseSetting = apps.get_model('certificates', 'CertificateGenerationCourseSetting')
+    CertificateGenerationCourseSetting.objects.all().update(self_generation_enabled=F('enabled'))
+
+class Migration(migrations.Migration):
+    """
+    Adds new field 'language_specific_templates_enabled'.
+    Also adds field 'self_generation_enabled' which is a 
+    replacement for 'enabled'
+    Lastly, copies data from 'enabled' to 'self_generation_enabled'
+    """
+    dependencies = [
+        ('certificates', '0008_schema__remove_badges'),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='certificategenerationcoursesetting',
+            name='language_specific_templates_enabled',
+            field=models.BooleanField(default=False, help_text="Render translated certificates rather than using the platform's default language. Available translations are controlled by the certificate template."),
+        ),
+        migrations.AddField(
+            model_name='certificategenerationcoursesetting',
+            name='self_generation_enabled',
+            field=models.BooleanField(default=False, help_text='Allow students to generate their own certificates for the course. Enabling this does NOT affect usage of the management command used for batch certificate generation.'),
+        ),
+        migrations.AlterField(
+            model_name='certificategenerationcoursesetting',
+            name='enabled',
+            field=models.BooleanField(default=False, help_text='DEPRECATED, please use self_generation_enabled instead.'),
+        ),
+        migrations.RunPython(copy_field, reverse_code=migrations.RunPython.noop),
```

Comment:
Missed this earlier, but have we tested how long this will take to run this copy with production size data(2k rows).  I don't expect it to take long but just want to make sure the due diligence was done.

Also, I agree, this definitely needs a real backwards migration.  If we have to role this migration back after updates were made(courses enabled certs).  We wouldn't want to break them on the way back.  We should copy data in both directions.

Reply:
Ah my bad, I thought that @clintonb was implying the reverse should be a noop.
"Add a reverse function, which should be RunSQL.noop"
I'll add a backwards copy instead.

Reply:
I have NOT yet tested how long this will take to run with production size data. What is a good approach for doing that?

Reply:
I *did* recommend using `noop`, but I understand @feanil's reasoning for using a real reverse migration.

Reply:
You should be able to do a simple test on your devstack.

1. Generate some data.(2k rows)
2. Run your migration(time it)
3. Run your backwards migration(time it)

If it's really slow there, we can move towards systems that are more like production but if it's pretty fast there, I think that we will likely be fine in production.

Reply:
(thumbsup)